langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -146,6 +149,45 @@ public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedde
         return addAll(null, embeddings, embedded);
     }
 
+    @Override
+    public void remove(String id) {
+        ensureNotBlank(id, "id");
+        client.data().deleter().
+                withClassName(objectClass).
+                withID(id).
+                run();
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids) {
+        ensureNotEmpty(ids, "ids");
+        client.batch().objectsBatchDeleter()
+                .withClassName(objectClass)
+                .withWhere(WhereFilter.builder()
+                        .path("id")
+                        .operator(Operator.ContainsAny)
+                        .valueText(ids.toArray(new String[0]))
+                        .build())
+                .run();
+    }
+
+    @Override
+    public void removeAll(Filter filter) {
+        ensureNotNull(filter, "filter");
+        WhereFilter whereFilter = WeaviateMetadataFilterMapper.map(filter);
```

Comment:
I suspect that the problem is that the filter path is "id" but probably should be "_metadata.id" or something like that. Metadata fields are stored not under the root, but under the `_metadata` field

Reply:
[https://github.com/kiratkumar47/langchain4j/blob/3c1a30fccd9c6c69f0d8268ecc5da9b0bf3ec7bc/langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateMetadataFilterMapper.java#L41-L47]
I have tried the following parameters in the path, but none of these seem to be working.
1. path("_metadata" + isEqualTo.key())
2. path("/_metadata/" + isEqualTo.key())
3. path("_metadata").path(isEqualTo.key())

![image](https://github.com/langchain4j/langchain4j/assets/41340746/34e5fef2-4f8a-4ec5-8e87-4935424daee8)
This is the definition of path in Documentation

Reply:
Did you try `path(METADATA, isEqualTo.key())` ?

Reply:
METADATA = "_metadata" (just in case)

Reply:
No working.

Reply:
I have created a free weaviate cloud cluster and here is what I see:

![image](https://github.com/langchain4j/langchain4j/assets/132277850/d38c1bcb-8a0a-49ed-bd17-9f57a7863594)

It seems that metadata field **_(and it's subfields)_** should be filterable for the test to work.

Reply:
How can I make it Filterable? I couldn't find anything in the documentation.

Reply:
Maybe you can ask here: https://forum.weaviate.io/

Reply:
Yes, I have already posted this issue on both the Weaviate form and StackOverflow.